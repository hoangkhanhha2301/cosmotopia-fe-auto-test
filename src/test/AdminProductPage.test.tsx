import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import AdminProductPage from '@/test/pages/AdminProductPage';

const products = [
  { id: '1', name: 'Áo khoác', price: 300000 },
  { id: '2', name: 'Giày thể thao', price: 500000 },
];

describe('AdminProductPage', () => {
  it('hiển thị sản phẩm và xử lý xoá', async () => {
    const handleDelete = jest.fn();
    render(<AdminProductPage products={products} onDelete={handleDelete} />);

    expect(screen.getByText(/áo khoác/i)).toBeInTheDocument();

    await userEvent.click(screen.getAllByText(/xoá/i)[0]);
    expect(handleDelete).toHaveBeenCalledWith('1');
  });

  it('hiển thị đúng tiêu đề và số dòng sản phẩm', () => {
  render(<AdminProductPage products={products} onDelete={jest.fn()} />);

  expect(screen.getByRole('heading', { name: /quản lý sản phẩm/i })).toBeInTheDocument();

  // Kiểm tra số dòng sản phẩm hiển thị
  const rows = screen.getAllByRole('row');
  expect(rows).toHaveLength(2); // 2 sản phẩm tương ứng 2 dòng (nếu không có <thead>)
});

it('hiển thị đúng tên và giá của từng sản phẩm', () => {
  render(<AdminProductPage products={products} onDelete={jest.fn()} />);

  expect(screen.getByText('Áo khoác')).toBeInTheDocument();
  expect(screen.getByText('300000')).toBeInTheDocument();
  expect(screen.getByText('Giày thể thao')).toBeInTheDocument();
  expect(screen.getByText('500000')).toBeInTheDocument();
});

it('gọi đúng onDelete khi bấm nút Xoá sản phẩm thứ 2', async () => {
  const handleDelete = jest.fn();
  render(<AdminProductPage products={products} onDelete={handleDelete} />);

  await userEvent.click(screen.getAllByText(/xoá/i)[1]); // sản phẩm 2

  expect(handleDelete).toHaveBeenCalledWith('2');
});

it('khớp với snapshot giao diện admin', () => {
  const { container } = render(<AdminProductPage products={products} onDelete={() => {}} />);
  expect(container).toMatchSnapshot();
});

});
