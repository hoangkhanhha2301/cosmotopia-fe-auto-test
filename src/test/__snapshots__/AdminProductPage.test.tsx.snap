// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminProductPage khớp với snapshot giao diện admin 1`] = `
<div>
  <div>
    <h1>
      Quản lý sản phẩm
    </h1>
    <table>
      <tbody>
        <tr>
          <td>
            Áo khoác
          </td>
          <td>
            300000
          </td>
          <td>
            <button>
              Xoá
            </button>
          </td>
        </tr>
        <tr>
          <td>
            Giày thể thao
          </td>
          <td>
            500000
          </td>
          <td>
            <button>
              Xoá
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
